crates/rust-analyzer/src/config.rs

```diff
@@ -1273,6 +1276,7 @@ impl Config {
         let sysroot_query_metadata = self.data.cargo_sysrootQueryMetadata;
 
         CargoConfig {
+            all_targets: self.data.cargo_allTargets,
```

Comment:
If `check_allTargets` is intended to overwrite `cargo_allTargets`, then that would have to be done here as well.